packages/yew-macro/src/html_tree/html_element.rs

```diff
@@ -135,39 +135,58 @@ impl ToTokens for HtmlElement {
         // other attributes
 
         let attributes = {
-            let normal_attrs = attributes.iter().map(|Prop { label, value, .. }| {
-                (label.to_lit_str(), value.optimize_literals_tagged())
-            });
-            let boolean_attrs = booleans.iter().filter_map(|Prop { label, value, .. }| {
-                let key = label.to_lit_str();
-                Some((
-                    key.clone(),
-                    match value {
-                        Expr::Lit(e) => match &e.lit {
-                            Lit::Bool(b) => Value::Static(if b.value {
-                                quote! { #key }
-                            } else {
-                                return None;
-                            }),
-                            _ => Value::Dynamic(quote_spanned! {value.span()=> {
-                                ::yew::utils::__ensure_type::<::std::primitive::bool>(#value);
-                                #key
-                            }}),
-                        },
-                        expr => Value::Dynamic(
-                            quote_spanned! {expr.span().resolved_at(Span::call_site())=>
-                                if #expr {
-                                    ::std::option::Option::Some(
-                                        ::yew::virtual_dom::AttrValue::Static(#key)
-                                    )
+            let normal_attrs = attributes.iter().map(
+                |Prop {
+                     label,
+                     value,
+                     directive,
+                     ..
+                 }| {
+                    (
+                        label.to_lit_str(),
```

Comment:
For properties, maybe we should apply a snake_case to camelCase conversion here like `web_sys`.
We only need to do this with `html!` macro as at runtime property names are set as `string`.

```rust
// snake case as we assume a it's like a field in Properties for an html element.
html! { <custom-element ~some_prop={"value"} /> };

// camel case as it's a string.
el.add_property("someProp", "value");
```

Reply:
I'm not sure if this is a good idea. We don't convert `snake_case` to `kebab-case` for attributes so this will be just another differentiator between the two. We can add it for both though, but that is for a future PR